packages/plugins/RSS3/src/SNSAdaptor/components/FeedCard/CollectibleCard.tsx

```diff
@@ -0,0 +1,259 @@
+                                components={{
+                                    user: <Label />,
+                                    cost: <Label />,
+                                    ens: verbose ? <Label /> : <span />,
+                                }}
+                            />
+                        ),
+                    }
+                }
+                metadata = (feed as RSS3BaseAPI.CollectibleTransferFeed).actions[0].metadata
+                return {
+                    metadata,
+                    summary: (
+                        <Translate.collectible_send
+                            values={{
+                                user,
+                                collectible: verbose ? metadata!.name : 'an NFT',
+                                recipient: formatEthereumAddress(feed.actions[0].address_to ?? '', 4),
+                            }}
+                            components={{
+                                user: <Label />,
+                                recipient: <Label />,
+                                collectible: verbose ? <Label /> : <span />,
+                            }}
+                        />
+                    ),
+                }
+            case Type.Burn:
+                metadata = (feed as RSS3BaseAPI.CollectibleBurnFeed).actions[0].metadata
+                return {
+                    metadata,
+                    summary: (
+                        <Translate.collectible_burn
+                            values={{
+                                user,
+                                collectible: verbose ? metadata!.name : 'an NFT',
+                            }}
+                            components={{
+                                user: <Label />,
+                                collectible: verbose ? <Label /> : <span />,
+                            }}
+                        />
+                    ),
+                }
+        }
+
+        return { summary: '' }
+    }, [feed, user])
+
+    const imageSize = verbose ? '100%' : 64
+
+    return (
+        <CardFrame type={CardType.TokenSwap} feed={feed} {...rest}>
+            <Typography className={classes.summary}>{summary}</Typography>
+            {metadata ? (
+                <div className={cx(classes.body, verbose ? classes.verbose : null)}>
+                    <Image
+                        classes={{ container: classes.image }}
+                        src={metadata.image}
+                        height={imageSize}
+                        width={imageSize}
+                    />
+                    <div className={classes.info}>
+                        {verbose ? null : <Typography className={classes.title}>{metadata.name}</Typography>}
+                        <Typography className={classes.subtitle}>{metadata.description}</Typography>
+                    </div>
+                    {verbose && 'attributes' in metadata && metadata.attributes?.length ? (
```

Comment:
```suggestion
                    {verbose && metadata.attributes?.length ? (
```

Reply:
type guarding here.

metadata could be `MintMetadata | TradeMetadata | TransactionMetadata | PoapMetadata`,  not all of them have `attributes`.